wpilibNewCommands/src/test/native/cpp/frc2/command/DeferredCommandTest.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include "CommandTestBase.h"
+#include "frc2/command/Commands.h"
+#include "frc2/command/DeferredCommand.h"
+
+using namespace frc2;
+
+class DeferredCommandTest : public CommandTestBase {
+ public:
+  void DeferredFunctionsTest(bool interrupted) {
+    std::unique_ptr<MockCommand> innerHolder = std::make_unique<MockCommand>();
+    MockCommand* innerCommand = innerHolder.get();
+    DeferredCommand deferred{[innerCommand] { return innerCommand; }, {}};
+
+    EXPECT_CALL(*innerCommand, Initialize());
+    EXPECT_CALL(*innerCommand, Execute());
+    EXPECT_CALL(*innerCommand, End(interrupted));
+
+    deferred.Initialize();
+    deferred.Execute();
+    EXPECT_FALSE(deferred.IsFinished());
+    innerCommand->SetFinished(true);
+    EXPECT_TRUE(deferred.IsFinished());
+    deferred.End(interrupted);
+  }
+};
+
+TEST_F(DeferredCommandTest, DeferredFunctions) {
+  DeferredFunctionsTest(false);
+  DeferredFunctionsTest(true);
+}
```

Comment:
Couldn't `DeferredCommandTest` be changed to inherit from `CommandTestBaseWithParam<bool>` instead of `CommandTestBase` and this be changed to `TEST_P`? (See http://google.github.io/googletest/advanced.html#value-parameterized-tests)

Reply:
I guess if it's felt strongly about. I didn't want to go through the setup for that one easy to duplicate test

Reply:
Parameterized tests feel like the correct solution, but as long as it's obvious which of the two failed - I don't mind. That isn't the situation with the current test code.